project/sherLOCKED/Assets/Scripts/Objects/DoorInteraction.cs

```diff
@@ -20,22 +24,90 @@ void OnTriggerEnter2D(Collider2D other)
+                SceneManager.LoadScene("Library");           
+                playerPos = transform.position;
+                playerPos.x = 4;
+                playerPos.y = 2;
+                other.gameObject.transform.position = playerPos;
+
+                // attempting to get sherry facing foward -> for future reference
+                // facing.SetFloat("input_y", -facing.GetFloat("input_y"));
+                // Time.timeScale = 1;
+
+            }
+            else if(roomName == "Hall" && this.name == "dining_door")
             {
-                SceneManager.LoadScene("DiningRoom");
+                SceneManager.LoadScene("DiningRoom");           
+                playerPos = transform.position;
+                playerPos.x = -3;
+                playerPos.y = 2;
+                other.gameObject.transform.position = playerPos;
+            }
+            else if(roomName == "DiningRoom" && this.name == "hall_door")
+            {
+                SceneManager.LoadScene("Hall");          
+                playerPos = transform.position;
+                playerPos.x = 4;
+                playerPos.y = 3;
+                other.gameObject.transform.position = playerPos;
+
+                // attempting to get sherry facing foward -> for future reference
+                // facing.SetFloat("input_y", -facing.GetFloat("input_y"));
+                // Time.timeScale = 1; 
             }
-            else if(roomName == "DiningRoom")
+            else if(roomName == "DiningRoom" && this.name == "ballroom_door")
             {
-                SceneManager.LoadScene("Hall");
+                SceneManager.LoadScene("Ballroom");          
+                playerPos = transform.position;
+                playerPos.x = 3;
+                playerPos.y = 2;
+                other.gameObject.transform.position = playerPos;
+
+                // attempting to get sherry facing foward -> for future reference
+                // facing.SetFloat("input_y", -facing.GetFloat("input_y"));
+                // Time.timeScale = 1; 
+            }
+            else if(roomName == "Library" && this.name == "hall_door")
+            {
+                SceneManager.LoadScene("Hall");          
+                playerPos = transform.position;
+                playerPos.x = -2;
+                playerPos.y = 3;
+                other.gameObject.transform.position = playerPos;
+            }
+            else if(roomName == "Ballroom" && this.name == "dining_door")
+            {
+                SceneManager.LoadScene("DiningRoom");          
+                playerPos = transform.position;
+                playerPos.x = 3;
+                playerPos.y = 2;
+                other.gameObject.transform.position = playerPos;
+            }
+
+
+            // Only destroy the player when loading the new scene if the player already exists
+            if(numOfPlayers == 1) {
+                DontDestroyOnLoad(other.gameObject);
```

Comment:
A few other things may also need adding to the Don'tDestroyOnLoad, namely the InteractionSymbol, and the ProgressCanvas

Reply:
Only works for root objects, so not destroying the canvas and it works